components/arm/fake/fake.go

```diff
@@ -65,30 +106,18 @@ func init() {
 
 // NewArm returns a new fake arm.
 func NewArm(cfg config.Component, logger golog.Logger) (arm.LocalArm, error) {
-	var model referenceframe.Model
-	var err error
+	var (
+		model referenceframe.Model
+		err   error
+	)
 	if cfg.ConvertedAttributes != nil {
-		converted := cfg.ConvertedAttributes.(*AttrConfig)
-
-		if converted.FailNew {
-			return nil, errors.New("whoops! failed to start up")
-		}
-
-		switch resource.ModelName(converted.ArmModel) {
-		case xarm.ModelName6DOF.Name:
-			model, err = xarm.Model(cfg.Name, 6)
-		case xarm.ModelName7DOF.Name:
-			model, err = xarm.Model(cfg.Name, 7)
-		case ur.ModelName.Name:
-			model, err = ur.Model(cfg.Name)
-		case yahboom.ModelName.Name:
-			model, err = yahboom.Model(cfg.Name)
-		case eva.ModelName.Name:
-			model, err = eva.Model(cfg.Name)
-		case ModelName.Name, "":
-			model, err = referenceframe.UnmarshalModelJSON(fakeModelJSON, cfg.Name)
-		default:
-			return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", cfg.ConvertedAttributes.(*AttrConfig).ArmModel)
+		modelPath := cfg.ConvertedAttributes.(*AttrConfig).ModelPath
+		armModel := cfg.ConvertedAttributes.(*AttrConfig).ArmModel
+		switch {
+		case armModel != "":
+			model, err = modelFromName(cfg.ConvertedAttributes.(*AttrConfig).ArmModel, cfg.Name)
+		case modelPath != "":
+			model, err = referenceframe.ModelFromPath(modelPath, cfg.Name)
 		}
 	} else {
```

Comment:
`Validate` throws an error if `ArmModel == "" && ModelPath == ""`. Hence, I do not think the `else` will ever be hit. I propose removing this.